internal/random/random.go

```diff
@@ -7,23 +7,31 @@ package random
 import (
 	crand "crypto/rand"
 	"encoding/binary"
-	"math/rand"
+	mrand "math/rand"
 
 	"go.opentelemetry.io/otel/trace"
+	erand "golang.org/x/exp/rand"
 )
 
 // SeedGlobal can be used to initialize the global thread-safe pseudo-random
 // generator from the standard library. This should be called from all init()
 // functions for all binaries.
+//
+// It seeds both math/rand and e/exp/rand packages just in case IDE imports
+// one or the other by accident.
 func SeedGlobal() {
-	var rngSeed int64
-	_ = binary.Read(crand.Reader, binary.LittleEndian, &rngSeed)
-	rand.Seed(rngSeed)
+	var seedInt64 int64
+	_ = binary.Read(crand.Reader, binary.LittleEndian, &seedInt64)
+	mrand.Seed(seedInt64)
+
+	var seedUint64 uint64
+	_ = binary.Read(crand.Reader, binary.LittleEndian, &seedUint64)
+	erand.Seed(seedUint64)
 }
```

Comment:
This is a defensive change - we already have x/exp/rand in our dependencies and GoLand tends to prefer this package or it was just a misclick by me. But using unseeded random generator is dangerous.

We can remove seeding completely with 1.20 Go upgrade.

Reply:
Can you explain the need for that random generator? I am not familiar with that.

Reply:
I meant seed as in [PRNG](https://en.wikipedia.org/wiki/Pseudorandom_number_generator).